docs/tutorials/typescript/work-queue-slack-app/index.mdx

```diff
@@ -0,0 +1,991 @@
+
+## Create TypeScript projects for the Slack App
+
+You need two TypeScript projects for this Slack application.
+You may have already created the first one for the Temporal Application in the [Prerequisite steps](/getting_started/typescript/dev_environment/).
+Here, make sure you have the second one for the Slack bot logic.
+
+This tutorial assumes that your TypeScript projects will be organized like this:
+
+```plaintext
+-- your-workqueue-slack-app
+  |-- temporal-application # This is the Temporal Application project
+  |   |-- .env
+  |   |-- src
+  |   |   |-- workflows
+  |   |   |   |-- workqueue.ts
+  |   |   |-- index.ts
+  |   |-- package.json
+  |   |-- tsconfig.json
+  |-- bot # This is the Slack bot project
+  |   |-- .env
+  |   |-- src
+  |   |   |-- index.ts
+  |   |-- package.json
+  |   |-- tsconfig.json
+  |-- common-types # These types are shared by both projects
+      |-- types.ts
+```
+
+This tutorial uses `npm` as the package manager.
+
+```
+npm init
+```
+
+And ensures there is a `tsconfig.json` file in each project root.
+
+```
+tsc --init
+```
+
+### Dependencies for the projects
+
+**Temporal Application dependencies**
+
+In the `temporal-application` project, you will need the following dependencies in your `package.json`:
+
+```json
+{
+  // ...
+  "dependencies": {
+    "@temporalio/worker": "^1.10.1",
+    "dotenv": "^16.1.3",
+    "path": "^0.12.7"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.7",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.4.5"
+  }
+  // ...
+}
+```
+
+**Slack bot dependencies**
+
+In the `bot` project, you will need the following dependencies in your `package.json`:
+
+```json
+{
+  // ...
+  "dependencies": {
+    "@slack/bolt": "^3.19.0",
+    "@slack/web-api": "^6.12.0",
+    "@temporalio/client": "^1.7.4",
+    "crypto": "^1.0.1",
+    "date-fns": "^3.6.0",
+    "dotenv": "^16.0.3"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.7",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.4.5"
+  }
+  // ...
+}
+```
+
+### Environment variables
+
+First, make sure you have `.env` added to your `.gitignore` file to avoid committing your credentials.
+
+**Slack bot environment variables**
+
+Grab the [Slack credentials](#credentials) you got from the Slack App configuration and add them to a `.env` file in the `bot` project:
+
+```plaintext
+SLACK_SIGNING_SECRET="<slack-signing-secret>"
+SLACK_BOT_TOKEN="<slack-bot-token>"
+SLACK_APP_TOKEN="<slack_app_token>"
+SLACK_WORKSPACE="<slack_workspace>"
+ENV="dev"
+```
+
+Also add your Slack Workspace and an `ENV` variable.
+The `ENV` variable is prepended to the Task Queue name for your Temporal Application.
+This enables you to use the same Temporal Namespace locally and in production without worrying about production Tasks getting executed on your local machine.
+
+**Temporal Application environment variables**
```

Comment:
Remove and make the below sentence more strongly worded (see above) -- we can also look into different color coding for when you have to run commands or edit configs in different environments, I find that helpful.